src/components/pages/userCoursesListPage.tsx

```diff
@@ -1,7 +1,79 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { UserCourse, Course } from 'devu-shared-modules'
 
 import PageWrapper from 'components/shared/layouts/pageWrapper'
+import UserCourseListItem from 'components/listItems/userCourseListItem'
+import Dropdown from 'components/shared/inputs/dropdown'
 
-const UserCoursesListPage = ({}) => <PageWrapper>User Courses List</PageWrapper>
+import RequestService from 'services/request.service'
+
+import styles from './userCoursesListPage.scss'
+
+const filterOptions = [
+  { label: 'All', value: 'all' },
+  { label: 'Active', value: 'active' },
+  { label: 'Inactive', value: 'inactive' },
+  { label: 'Dropped', value: 'dropped' },
+]
+
+const UserCoursesListPage = () => {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const [userCourses, setUserCourses] = useState(new Array<UserCourse>())
+  const [courses, setCourses] = useState<Record<string, Course>>({})
+  const currentFilter = JSON.parse(localStorage.getItem('status') || '')
+  const [filter, setFilter] = useState<'all' | 'active' | 'inactive' | 'dropped'>(currentFilter)
+
+  const fetchData = async () => {
+    try {
+      // The filter isn't implemented by the API yet
+      const userCourses: UserCourse[] = await RequestService.get(`/api/user-courses?filterBy=${filter}`)
+      const courseRequests = userCourses.map((u) => RequestService.get(`/api/courses/${u.courseId}`))
+      const courses: Course[] = await Promise.all(courseRequests)
+
+      localStorage.setItem('status', JSON.stringify(filter) || '')
+
+      // Mapify course ids so we can look them up more easilly via their id
+      const courseMap: Record<string, Course> = {}
+      for (const course of courses) courseMap[course.id || ''] = course
+
+      setUserCourses(userCourses)
+      setCourses(courseMap)
+    } catch (error) {
+      setError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [filter])
+
+  if (loading) return <div>Loading</div>
+  if (error) return <div>{error}</div>
```

Comment:
This is a bit out of scope of what you've done here, but I think it'd be good to add:

We should make an error page that can very cleanly display these errors. You'll find a pretty good example with the `<ValidationErrorViewer />` component. The difference here is make it a full page. So wrap it with it's own `<PageWrapper />` and have it take an error object and display it just like that validation error viewer. This way we can reuse it as an error page basically everywhere.

Reply:
https://user-images.githubusercontent.com/56481968/127404595-5aaef1e0-0c6b-4654-b0b9-71cb4bb767e4.mov

Something Like this? I made a commit with my most recent changes.

Reply:
If you're providing the name then sure. We can work on it. But it's good enough. We don't have a name though at least from the error response. So you'd need to provide a title then display the message that comes back with the response

Reply:
In that case I'll remove the Error Name section and leave the error message at the bottom